python/cudf/cudf/core/udf/groupby_function.py

```diff
@@ -0,0 +1,867 @@
+
+
+@register_model(GroupType)
+class GroupModel(models.StructModel):
+    def __init__(
+        self, dmm, fe_type
+    ):  # fe_type is fully instantiated group type
+        members = [
+            ("group_data", types.CPointer(fe_type.group_scalar_type)),
+            ("size", types.int64),
+            ("index", types.CPointer(fe_type.index_type)),
+        ]
+        models.StructModel.__init__(self, dmm, fe_type, members)
+
+
+my_max_int32 = cuda.declare_device(
+    "BlockMax_int32", "types.int32(types.CPointer(types.int32),types.int64)"
+)
+
+my_max_int64 = cuda.declare_device(
+    "BlockMax_int64", "types.int64(types.CPointer(types.int64),types.int64)"
+)
+
+my_max_float64 = cuda.declare_device(
+    "BlockMax_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_min_int32 = cuda.declare_device(
+    "BlockMin_int32", "types.int32(types.CPointer(types.int32),types.int64)"
+)
+
+my_min_int64 = cuda.declare_device(
+    "BlockMin_int64", "types.int64(types.CPointer(types.int64),types.int64)"
+)
+
+my_min_float64 = cuda.declare_device(
+    "BlockMin_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_sum_int64 = cuda.declare_device(
+    "BlockSum_int64", "types.int64(types.CPointer(types.int64),types.int64)"
+)
+
+my_sum_float64 = cuda.declare_device(
+    "BlockSum_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_mean_int64 = cuda.declare_device(
+    "BlockMean_int64",
+    "types.float64(types.CPointer(types.int64),types.int64)",
+)
+
+my_mean_float64 = cuda.declare_device(
+    "BlockMean_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_std_int64 = cuda.declare_device(
+    "BlockStd_int64",
+    "types.float64(types.CPointer(types.int64),types.int64)",
+)
+
+my_std_float64 = cuda.declare_device(
+    "BlockStd_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_var_int64 = cuda.declare_device(
+    "BlockVar_int64",
+    "types.float64(types.CPointer(types.int64),types.int64)",
+)
+
+my_var_float64 = cuda.declare_device(
+    "BlockVar_float64",
+    "types.float64(types.CPointer(types.float64),types.int64)",
+)
+
+my_idxmax_int64 = cuda.declare_device(
+    "BlockIdxMax_int64",
+    "types.int64(types.CPointer(types.int64),"
+    "types.CPointer(types.int64),types.int64)",
+)
+
+my_idxmax_float64 = cuda.declare_device(
+    "BlockIdxMax_float64",
+    "types.int64(types.CPointer(types.float64),"
+    "types.CPointer(types.int64),types.int64)",
+)
+
+my_idxmin_int64 = cuda.declare_device(
+    "BlockIdxMin_int64",
+    "types.int64(types.CPointer(types.int64),"
+    "types.CPointer(types.int64),types.int64)",
+)
+
+my_idxmin_float64 = cuda.declare_device(
+    "BlockIdxMin_float64",
+    "types.int64(types.CPointer(types.float64),"
+    "types.CPointer(types.int64),types.int64)",
+)
+
+# Load the highest compute capability file available that is less than
```

Comment:
This logic has changed significantly and should be refactored into a utility so that `strings_udf` and this groupby apply machinery can both use it. See my comment about breaking up this file below.